engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java

```diff
@@ -1720,6 +1720,9 @@ private Pair<VolumeVO, DataStore> recreateVolume(VolumeVO vol, VirtualMachinePro
             DiskOffering diskOffering = _entityMgr.findById(DiskOffering.class, vol.getDiskOfferingId());
             if (diskOffering.getEncrypt()) {
                 vol = setPassphraseForVolumeEncryption(vol);
+            } else if (vol.getPassphraseId() != null) {
```

Comment:
@vishesh92 I think, better to not allow offering change for encrypted volumes if disk offering doesn't support encryption.